src/macaron/artifact/local_artifact.py

```diff
@@ -0,0 +1,275 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module declares types and utilities for handling local artifacts."""
+
+import fnmatch
+import glob
+import os
+from collections.abc import Mapping
+
+from packageurl import PackageURL
+
+from macaron.artifact.maven import construct_maven_repository_path
+
+
+def construct_local_artifact_paths_glob_pattern_maven_purl(maven_purl: PackageURL) -> list[str] | None:
+    """Return a list of glob pattern(s) representing maven artifacts in a local maven repository.
+
+    The glob pattern(s) can be used to search in `<...>/.m2/repository` directory.
+
+    Parameters
+    ----------
+    maven_purl : PackageURL
+        A maven type PackageURL instance.
+
+    Returns
+    -------
+    list[str] | None
+        A list of glob patterns or None if an error happened.
+
+    Examples
+    --------
+    >>> from packageurl import PackageURL
+    >>> purl = PackageURL.from_string("pkg:maven/com.oracle.macaron/macaron@0.13.0")
+    >>> construct_local_artifact_paths_glob_pattern_maven_purl(purl)
+    ['com/oracle/macaron/macaron/0.13.0']
+    """
+    if not maven_purl.type == "maven":
```

Comment:
```suggestion
    if maven_purl.type != "maven":
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/864/commits/bd5eeaa78fde8fe25b550ec08ab74815aab3c04f